tests/utils/common/common.py

```diff
@@ -126,23 +126,23 @@ class ReadFileLock(filelock.BaseFileLock):
 
     def _acquire(self):
         open_mode = os.O_RDWR | os.O_CREAT | os.O_TRUNC
-        fd = os.open(self._lock_file, open_mode)
+        fd = os.open(self._context.lock_file, open_mode)
 
         try:
             fcntl.flock(fd, fcntl.LOCK_SH | fcntl.LOCK_NB)
         except (IOError, OSError):
             os.close(fd)
         else:
-            self._lock_file_fd = fd
+            self._context._lock_file_fd = fd
```

Comment:
@kacf There is a bug (here and below),

```suggestion
            self._context.lock_file_fd = fd
```

I will instead refresh the whole code from upstream so that the comment above remains true (that the only difference is LOCK_SH instead of LOCK_EX) and we ease future maintenance.

Reply:
Good idea.